### PR TITLE
Fix 'invert if' mishandling of lifted nullable operators

### LINKED ISSUE #63311: 'Invert if' mishandles lifted relational operators
**Version Used**: 17.4.0-p1

```cs
using System;

class C
{
    void M(Point? point)
    {
        if (point?.X < 10)
        {
            return;
        }

        Console.WriteLine("Point is null or X >= 10");
    }
}

struct Point
{
    public int X { get; set; }
    public int Y { get; set; }
}
```

💡 Invert if:

```cs
    void M(Point? point)
    {
        if ((point?.X) >= 10)
        {
            // ❌ The logic has changed and is now "point is not null and X >= 10"
            // Thus, this message is now untrue
            Console.WriteLine("Point is null or X >= 10");
        }
    }
```

Expected:

```cs

    void M(Point? point)
    {
        if (!(point?.X < 10))
        {
            Console.WriteLine("Point is null or X >= 10");
        }
    }
```

### PR DESCRIPTION
**Pull request by jnm2**
Fixes #63311

### CODE REVIEW

File: src/EditorFeatures/CSharpTest/InvertIf/InvertIfTests.cs [hunk 1]
```diff
@@ -1291,5 +1291,174 @@ int M()
...
+        [Fact, WorkItem(63311, "https://github.com/dotnet/roslyn/issues/63311")]
+        public async Task TestLiftedNullable_GreaterThan()
+        {
+            await TestInRegularAndScriptAsync(
+                """
+                class C
+                {
+                    void M(int? p)
+                    {
+                        [||]if (p > 10)
```
Comment (DoctorKrolic) [inline src/EditorFeatures/CSharpTest/InvertIf/InvertIfTests.cs:1304]: Maybe create a `[Theory]` and use operator as a test parameter?

**Review by CyrusNajmabadi — APPROVED** [review]

**Comment by CyrusNajmabadi** [comment]
Thanks djinnm2!

**Comment by DoctorKrolic** [comment]
Ok, I got ignored...

**Comment by CyrusNajmabadi** [comment]
@DoctorKrolic you weren't ignored. I didn't see your post when I reviewed, and this was set to auto merge. I have no idea if jnm saw it either as he has done nothing else on this pr. 

I will say that in generally not a fan of theories changing test input or output data. I personally find it often harder to read and I'm totally fine with individual test cases.

**Comment by jnm2** [comment]
It looks like Cyrus approved 4 minutes after @DoctorKrolic commented, and since these pages don't live-update with comments, Cyrus probably had initially clicked the link to open the PR just prior to that. This happens a lot and is probably the most common paper cut I've had with GitHub.

File: src/EditorFeatures/CSharpTest/InvertIf/InvertIfTests.cs (hunk 1 above)
Comment (jnm2) [reply src/EditorFeatures/CSharpTest/InvertIf/InvertIfTests.cs:1304]: I decided against this originally based on looking at nearby tests.

**Pull request merged**